content/en/blog/2025/announcing-etcd-3.6.md

```diff
@@ -0,0 +1,425 @@
+
+- Removed `etcdctl snapshot status`
+
+  `etcdctl` no longer supports retrieving the status of a snapshot. Use the `etcdutl snapshot status` command instead.
+
+- Removed `etcdctl snapshot restore`
+
+  `etcdctl` no longer supports restoring from a snapshot. Use the `etcdutl snapshot restore` command instead.
+
+## Critical bug fixes
+
+Correctness has always been a top priority for the etcd project. In the process of developing 3.6.0, we found and
+fixed a few notable bugs that could lead to data inconsistency in specific cases. These fixes have been backported
+to previous releases, but we believe they deserve special mention here.
+
+- Data Inconsistency when Crashing Under Load
+
+Previously, when etcd was applying data, it would update the consistent-index first, followed by committing the
+data. However, these operations were not atomic. If etcd crashed in between, it could lead to data inconsistency
+(see [issue/13766][]). The issue was introduced in v3.5.0, and fixed in v3.5.3 with [PR/13854][].
+
+- Durability API guarantee broken in single node cluster
+
+When a client writes data and receives a success response, the data is expected to be persisted. However, the data might
+be lost if etcd crashes immediately after sending the success response to the client. This was a legacy issue (see [issue/14370][])
+affecting all previous releases. It was addressed in v3.4.21 and v3.5.5 with [PR/14400][], and fixed in raft side in
+main branch (now release-3.6) with [PR/14413][].
+
+- Revision Inconsistency when Crashing During Defragmentation
+
+If etcd crashed during the defragmentation operation, upon restart, it might reapply
+some entries which had already been applied, accordingly leading to the revision inconsistency issue
+(see the discussions in [PR/14685][]). The issue was introduced in v3.5.0, and fixed in v3.5.6 with [PR/14730][].
+
+## Upgrade issue
+
+This section highlights a common issue [issues/19557][] in the etcd v3.5 to v3.6 upgrade that may cause the upgrade
+process to fail. For a complete upgrade guide, refer to [Upgrade etcd from v3.5 to v3.6][].
+
+The issue was introduced in etcd v3.5.1, and resolved in v3.5.20.
+
+**Key takeaway**: users are required to first upgrade to etcd v3.5.20 (or a higher patch version) before upgrading
+to etcd v3.6.0; otherwise, the upgrade may fail.
+
+For more background and technical context, see [upgrade_from_3.5_to_3.6_issue][].
+
+## Testing
+
+We introduced the [Robustness testing][] to verify correctness, which has always been our top priority.
+It plays traffic of various types and volumes against an etcd cluster, concurrently injects a random
+failpoint, records all operations (including both requests and responses), and finally performs a
+linearizability check. It also verifies that the [Watch APIs][] guarantees have not been violated.
+The robustness test increases our confidence in ensuring the quality of each etcd release.
```

Comment:
I would one important point here. That this testing approach not only found issues in v3.5, but long standing issues preceding v3.4 release. Meaning etcd v3.6 might be the most bug free release in the history of the project.

Reply:
This was touched more in https://opensource.googleblog.com/2024/06/driving-etcd-stability-and-kubernetes-success.html